ext/SpeedyWeatherCUDAExt/fourier.jl

```diff
@@ -0,0 +1,119 @@
+
+# Override FFT package deciding function
+SpeedyTransforms.which_FFT_package(::Type{<:CuArray{<:AbstractFloat}}) = CUFFT 
+
+"""$(TYPEDSIGNATURES)
+Util function to generate FFT plans based on the array type of the fake Grid 
+data provided. Uses indexing as we seemingly can't use views for FFT planning 
+with CUFFT."""
+function SpeedyTransforms.plan_FFTs!(
+    rfft_plans::Vector{AbstractFFTs.Plan},
+    brfft_plans::Vector{AbstractFFTs.Plan},
+    rfft_plans_1D::Vector{AbstractFFTs.Plan},
+    brfft_plans_1D::Vector{AbstractFFTs.Plan},
+    fake_grid_data::AbstractGridArray{NF, N, <:CuArray{NF}},
+    scratch_memory_north::CuArray{Complex{NF}},
+    rings::AbstractArray,
+    nlons::Vector{<:Int}
+) where {NF<:AbstractFloat, N}
+    # Determine which FFT package to use (currently either FFTW or GenericFFT)
+    FFT_package = SpeedyTransforms.which_FFT_package(CuArray{NF})
+
+    # For each ring generate an FFT plan (for all layers and for a single layer)
+    for (j, nlon) in enumerate(nlons)
+        real_matrix_input = fake_grid_data.data[rings[j], :]
+        complex_matrix_input = scratch_memory_north[1:nlon÷2 + 1, :, j]
+        real_vector_input = fake_grid_data.data[rings[j], 1]
+        complex_vector_input = scratch_memory_north[1:nlon÷2 + 1, 1, j]
+
+        rfft_plans[j] = FFT_package.plan_rfft(real_matrix_input, 1)
+        brfft_plans[j] = FFT_package.plan_brfft(complex_matrix_input, nlon, 1)
+        rfft_plans_1D[j] = FFT_package.plan_rfft(real_vector_input, 1)
+        brfft_plans_1D[j] = FFT_package.plan_brfft(complex_vector_input, nlon, 1)
+    end
```

Comment:
I thought this but when I tested it seemed to break the current implementation of `fourier_batched!` as FFTW apparently allows planning on views and restricts those plans to only working on views. I will test again on Monday as in retrospect that does seem odd behaviour.

Reply:
Turns out I just needed to apply the plan on the same stride array in `fourier_batched!` rather than on a subarray, but this means allocating in the transform and slows things down a bit. I think it's probably worth the small amount of duplicated code for a the speedup and lower memory use?

Reply:
yes